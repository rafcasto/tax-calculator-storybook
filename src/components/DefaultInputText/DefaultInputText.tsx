import React from "react";
import TextField from '@mui/material/TextField';

interface DefaultInputTextProps {
    id?:string,
    label?:string,
    variant?: 'outlined' | 'filled' | 'standard' ,
    onchange?: (event: object) => void
} 

export const DefaultInputText = ({
    id,
    label,
    variant='standard',...props}:DefaultInputTextProps) => {
    return (<TextField  fullWidth  id={id} label={label} variant={variant} />)
} 