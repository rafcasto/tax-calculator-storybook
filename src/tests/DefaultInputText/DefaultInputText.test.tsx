import React from 'react'
import {render,screen} from '@testing-library/react' 
import renderer from 'react-test-renderer';
import {DefaultInputText} from '../../components/DefaultInputText/DefaultInputText'

test('render standard variant', () => {
    render (<DefaultInputText label='standard' id='standard'  />)
    const inputText = screen.getByTestId("standard");
    expect(inputText).toBeInTheDocument();
} );


test('snapshot test', () =>  {
   const inputText = renderer.create(<DefaultInputText label='outlined' id='outlined' variant='outlined' />)
   .toJSON()
   expect(inputText).toMatchSnapshot();
});