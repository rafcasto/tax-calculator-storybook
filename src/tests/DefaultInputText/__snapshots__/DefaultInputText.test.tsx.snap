// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  data-testid="outlined"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
    htmlFor="outlined"
    id="outlined-label"
  >
    outlined
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id="outlined"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-yjsfm1"
      >
        <span>
          outlined
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
